features/login_github_redirect.feature

```diff
@@ -0,0 +1,151 @@
+
+  As a user who first opens the app
+  So that I can efficiently login and browse the app
+  I want to return to the page I requested to login
+
+Background: users, orgs and apps have been added to database
+  Given the following apps exist:
+        | name  | description | org_id | status  |
+        | app1  | test        | 1      | pending |
+        | app2  | test        | 1      | pending |
+        | app3  | test        | 1      | pending |
+
+    And the following orgs exist:
+        | name | contact_id |
+        | org1 | 1          |
+        | org2 | 1          |
+        | org3 | 1          |
+
+    And the following users exist:
+        | name  | github_uid      | email         |
+        | user1 | esaas_developer | test@user.com |
+        | user2 |                 | test@user.com |
+        | user3 |                 | test@user.com |
+
+
+  And I am not logged in
+
+Scenario: not logged in so should see login with Github from users page
+  #Story ID: #152298585
+  Given I am on the users page
+  Then I should see "Log in with GitHub"
+ 
+Scenario: not logged in so should see login with Github from orgs page 
+  #Story ID: #152298585
+  Given I am on the orgs page
+  Then I should see "Log in with GitHub"
+  
+Scenario: not logged in so should see login with Github from create page 
+  #Story ID: #152298585
+  Given I am on the create page
+  Then I should be on the login page
+  # Then I should see "Log in with GitHub"
+
+Scenario: login with Github from orgs page and redirect to orgs
+  #Story ID: #152298585
+  Given I am not logged in
+  And I am on the apps page
+  And I follow "Orgs"
+  And I follow "Log in with GitHub"
+  Then I should be on the orgs page
+
+Scenario: login with Github from users page and redirect to users
+  #Story ID: #152298585
+  Given I am not logged in
+  And I am on the apps page
+  And I follow "Users"
+  #And I am visiting the provider "users"
+  And I follow "Log in with GitHub"
+  Then I should be on the users page
+
+Scenario: login with Github from create page and redirect to create
+  #Story ID: #152298585
+  Given I am not logged in
+  And I am on the apps page
+  Given I follow "Create"
+  And I follow "Log in with GitHub"
+  Then I should be on the create page
+  
+Scenario: login with Github from create page, logout, login back and redirect to create
+  #Story ID: #152298585
+  Given I am not logged in
+  And I am on the apps page
+  And I follow "Create"
+  And I follow "Log in with GitHub"
+  And I am not logged in
+  And I follow "Log in with GitHub"
+  Then I should be on the create page
+  
+Scenario: login with Github from create page, after visiting orgs and users
+  #Story ID: #152298585
+  Given I am not logged in
+  And I am on the apps page
+  And I follow "Create"
+  And I follow "Orgs"
+  And I follow "Users"
+  And I follow "Create"
+  And I follow "Log in with GitHub"
+  Then I should be on the create page
+  
+Scenario: Login to github to create a new app
```

Comment:
I see what you are trying to test: user wants to create a new app and when they log in to the app, they should be redirected to "New app" page. You indeed test this behavior in the first half of the scenario, but the second half seems to test the behavior of apps_controller#new and apps_controller#create. Keeping only the first half makes it a better scenario

Reply:
Thanks for the suggestion! I kept the login part of the test in the login feature and moved the apps_controller test part to a new feature for apps_controller.